Add host API for *_on_stream operations (#340)
ROCm-Developer-Tools/ROC_SHMEM: docs/api/rma.rst

**Description:**
This routine writes contiguous data of ``nelems`` bytes from source on the calling PE to ``dest`` at ``pe``.

ROCSHMEM_PUTMEM_ON_STREAM
--------------------------

> .. cpp:function:: __host__ void rocshmem_putmem_on_stream(void *dest, const void *source, size_t bytes, int pe, hipStream_t stream)

Reviewer: This API should be keep consistent with `rocshmem_putmem`. There are a few comments in the docs adjusting `bytes` to `nelems`, the same changes should be made to other APIs and the code itself.

I agree with what was done here with `bytes` (it makes more sense to me too) but I think its best to stick with convention

```suggestion
.. cpp:function:: __host__ void rocshmem_putmem_on_stream(void *dest, const void *source, size_t nelems, int pe, hipStream_t stream)
```

Author: Done

    :param dest:    Destination address. Must be an address on the symmetric heap.
    :param source:  Source address. Must be an address on the symmetric heap.

>     :param bytes:   Number of bytes to transfer.

Reviewer: ```suggestion
    :param nelems:   The number of elements to transfer.
```

Author: Done. I took the wording from `rocshmem_putmem`. It clarifies that `nelems` is in bytes

":param nelems: Size of the transfer in bytes."

    :param pe:      PE of the remote process.
    :param stream:  HIP stream on which to enqueue the operation.

    :returns:       None.

**Description:**
This routine enqueues a putmem RMA operation on a HIP stream. The function writes contiguous

> data of ``bytes`` bytes from source on the calling PE to ``dest`` at ``pe``. The operation

Reviewer: ```suggestion
data of ``nelems`` elements from source on the calling PE to ``dest`` at ``pe``. The operation
```

Author: In `ROCSHMEM_PUTMEM` description it states "``nelems`` bytes". I would prefer also to keep the same for `ROCSHMEM_PUTMEM_ON_STREAM` and others, because it clarifies for the customer that one element is one byte
